automation/prepare-release.py

```diff
@@ -36,7 +36,18 @@
 UNRELEASED_CHANGES_FILE = "CHANGES_UNRELEASED.md"
 CHANGELOG_FILE = "CHANGELOG.md"
 
-# 1. Calculate new version number.
+# 1. Create a new branch based on the branch we want to release from.
+
+ensure_working_tree_clean()
+
+step_msg(f"Updating remote {remote}")
+run_cmd_checked(["git", "remote", "update", remote])
+
+release_branch = f"cut-{next_version_full}"
```

Comment:
At this point `next_version_full` is still undefined, how about we create the branch first then rename it with `git branch -M <name>` after we determine the version number?

Reply:
good catch; updated